src/app/index.js

```diff
@@ -1,69 +1,208 @@
+import { signup } from '../app/signup/signup';
 
-  slides = document.getElementsByClassName("mySlides");
-  dots = document.getElementsByClassName("dot");
+const WINDOW_CONFIG = window['SABKABAZAR_CONFIG'];
 
-  content.innerHTML = routes['home'];
-  this.slideIndex = 1;
-  this.showSlides(this.slideIndex);
+let slideIndex, slides, dots, banners, categories, productList;
+
+/**============= Handling page change on navigation click keeping header/footer fixed =================== */
+const routes = {
+  'signup': signup,
+  'signin': login,
+  'home': home,
+  'products': products
+}
+
+$(document).ready(function() {
+
+  let handlebarsData;
+
+  // fetch banners and then categories
+  makeRequest(`${WINDOW_CONFIG.apiUrl}/mock-data/banners/banners.json`, 'GET')
+    .then(function(bannersData) {
+
+      makeRequest(`${WINDOW_CONFIG.apiUrl}/mock-data/categories/categories.json`, 'GET')
+        .then(function(categoriesData) {
+
+          banners = bannersData;
+          categories = categoriesData;
+          
+          slides = document.getElementsByClassName('bannerSlides');
+          dots = document.getElementsByClassName('dot');
+        
+          // creating data for handlebars
+          handlebarsData = {
+            banners, categories
+          }
+
+          // appending content section with home view(by default)
+          onNavItemClick('home', handlebarsData);
+        })
+        .catch(function (error) {
+          console.error('Something went wrong in categories', error);
+        });
+    })
+    .catch(function (error) {
+      console.error('Something went wrong in banners', error);
+    });
+
+  /**================== Navigation buttons ============================ */
+
+  $('.nav-home').on('click', function() {
+    handlebarsData = {
+      banners, categories
+    }
+    onNavItemClick('home', handlebarsData);
+    onInitSlider();
+    onDrawerCloseIconClick();
+  });
+
+  $('.nav-product').on('click', function() {
+    makeRequest(`${WINDOW_CONFIG.apiUrl}/mock-data/products/products.json`, 'GET')
+      .then(function(products) {
+
+        productList = products;
+
+        handlebarsData = {
+          products, categories
+        }
+
+        onClickProductOrCategory('products', handlebarsData);
+      })
+      .catch(function(error) {
+        console.error('Something went wrong in products', error);
+      })
+  });
+
+  $('.btn-signin').on('click', function() {
+    onNavItemClick('signin');
+    onDrawerCloseIconClick();
+  });
+
+  $('.btn-register').on('click', function() {
+    onNavItemClick('signup');
+    onDrawerCloseIconClick();
+  });
+
+  /**================== Login/Signup focus ============================ */
+  $('.input100').each(function(){
+    $(this).on('blur', function(){
+      if($(this).val().trim() != "") {
+        $(this).addClass('has-val');
+      }
+      else {
+        $(this).removeClass('has-val');
+      }
+    })    
+  });
+
+  let showPass = 0;
```

Comment:
Why you doing such hacky solutions? Even if you need it use Boolean for flags. Event attachment and non delegation is global problem